numpyro/distributions/mixtures.py

```diff
@@ -0,0 +1,203 @@
+        """
+        return self._component_distribution
+
+    @constraints.dependent_property
+    def support(self):
+        return self.component_distribution.support
+
+    @property
+    def is_discrete(self):
+        return self.component_distribution.is_discrete
+
+    def tree_flatten(self):
+        mixing_flat, mixing_aux = self.mixing_distribution.tree_flatten()
+        component_flat, component_aux = self.component_distribution.tree_flatten()
+        params = (mixing_flat, component_flat)
+        aux_data = (
+            (type(self.mixing_distribution), type(self.component_distribution)),
+            (mixing_aux, component_aux),
+        )
+        return params, aux_data
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, params):
+        mixing_params, component_params = params
+        child_clss, child_aux = aux_data
+        mixing_cls, component_cls = child_clss
+        mixing_aux, component_aux = child_aux
+        mixing_dist = mixing_cls.tree_unflatten(mixing_aux, mixing_params)
+        component_dist = component_cls.tree_unflatten(component_aux, component_params)
+        return cls(
+            mixing_distribution=mixing_dist, component_distribution=component_dist
+        )
+
+    @property
+    def mean(self):
+        probs = self.mixing_distribution.probs
+        probs = probs.reshape(probs.shape + (1,) * self.event_dim)
+        weighted_component_means = probs * self.component_distribution.mean
+        return jnp.sum(weighted_component_means, axis=self.mixture_dim)
+
+    @property
+    def variance(self):
+        probs = self.mixing_distribution.probs
+        probs = probs.reshape(probs.shape + (1,) * self.event_dim)
+        # E[Var(Y|X)]
+        mean_cond_var = jnp.sum(
+            probs * self.component_distribution.variance, axis=self.mixture_dim
+        )
+        # Variance is the expectation of the squared deviation of a random variable from its mean
+        sq_deviation = (
+            self.component_distribution.mean
+            - jnp.expand_dims(self.mean, axis=self.mixture_dim)
+        ) ** 2
+        # Var(E[Y|X])
+        var_cond_mean = jnp.sum(probs * sq_deviation, axis=self.mixture_dim)
+        # Law of total variance: Var(Y) = E[Var(Y|X)] + Var(E[Y|X])
+        return mean_cond_var + var_cond_mean
+
+    def cdf(self, samples):
+        """
+        The cumulative distribution function of this mixture distribution.
+
+        :param value: samples from this distribution.
+        :return: output of the cummulative distribution function evaluated at `value`.
+        :raises: NotImplementedError if the component distribution does not implement the cdf method.
+        """
+        cdf_components = self.component_distribution.cdf(
+            jnp.expand_dims(samples, axis=self.mixture_dim)
+        )
+        return jnp.sum(cdf_components * self.mixing_distribution.probs, axis=-1)
+
+    def sample_with_intermediates(self, key, sample_shape=()):
```

Comment:
Could you add docs mention what this method returns?

Reply:
Added a docstring mentioning the return tuple